system/parted.py

```diff
@@ -0,0 +1,169 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import locale
+import math
+
+
+DOCUMENTATION = '''
+---
+author: "Vladimir Kozhukalov"
+module: parted
+short_description: Configure block device partitions
+description:
+  - This module creates, removes block device partitions.
+version_added: "2.2"
+options:
+  device:
+    description:
+    - The device file. E.g. /dev/sda
+    required: true
+  align:
+    description:
+    - Alignment type for newly created partitions
+    required: false
+    choices: ["none", "cylinder", "minimal", "optimal"]
+  script:
+    description:
+    - List of parted script items
+    required: false
+'''
+
+
+RETURN = '''
+partition_info:
+  description: Current partition information
+  returned: success
+  type: dict
+  contains:
+    generic:
+      description: Generic device information
+      type: dict
+    partitions:
+      description: List of device partitions
+      type: list
+'''
+
+
+EXAMPLES = '''
+# Do nothing but get partition info.
+- parted:
+  device: /dev/sda
+
+# Create partitions on device /dev/sda.
+- parted:
+  device: /dev/sda
+  script:
+    - "mkpart primary 10 20"
+    - "mkpart primary 20 30"
+
+# Remove partition on device /dev/sda.
+- parted:
+  device: /dev/sda
+  script:
+    - "rm 1"
+
+# Remove all partitions on device /dev/sda.
+- parted:
+  device: /dev/sda
+  register: result
+- parted:
+  device: /dev/sda
+  script:
+    - "rm {{ item.num }}"
+  with_items: "{{ result.partition_info.partitions }}"
+
+
+'''
+
+
+def parse_unit(s, unit, ceil=True):
+    """Converts '123.1unit' string into ints
+
+    If ceil is True it will be rounded up (124)
+    and and down (123) if ceil is False.
+    """
+
+    flt = locale.atof(s.split(unit)[0])
+    if ceil:
+        return int(math.ceil(flt))
+    return int(math.floor(flt))
+
+
+def parse_partition_info(output):
+    lines = output.split('\n')
+    generic_params = lines[1].rstrip(';').split(':')
+    generic = {
+        'dev': generic_params[0],
+        'size': parse_unit(generic_params[1], 'MiB'),
+        'logical_block': int(generic_params[3]),
+        'physical_block': int(generic_params[4]),
+        'table': generic_params[5],
+        'model': generic_params[6]
+    }
+    parts = []
+    for line in lines[2:]:
+        line = line.strip().rstrip(';')
+        if not line:
+            continue
+        part_params = line.split(':')
+        parts.append({
+            'num': int(part_params[0]),
+            'begin': parse_unit(part_params[1], 'MiB'),
+            'end': parse_unit(part_params[2], 'MiB'),
+            'size': parse_unit(part_params[3], 'MiB'),
+            'fstype': part_params[4] or None
+        })
+    return {'generic': generic, 'partitions': parts}
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            device=dict(required=True),
+            align=dict(default='optimal'),
```

Comment:
add the choices list like you did in the arg docs:

~~~python
align=dict(default='optimal', choices=['none', 'cylinder', 'minimal', 'optimal']),
~~~